### DISCUSSION TITLE
Reloading glightbox not working as expected

### DISCUSSION THREAD

**Discussion by arminfraude** [biati-digital/glightbox#431] [Q&A]
Good day,

let's get straight to what I'm doing:
I am inserting _images_ into my _galleryLightbox_ object using Ajax calls or DOM manipulations. 
This is done at the head of the array _galleryLightbox.elements_.
According to the description of the methods here, calling _reload()_ on the lightbox object should then update it correctly:

> Reload the lightbox, after inserting content with ajax.

Unfortunately, this is not quite true. 
The array _galleryLightbox.elements_ is now correctly extended with the new images.
However, _galleryLightbox.slidesContainer_ is not updated correctly: 
A new slide is inserted at the end, but it is empty when I try to zoom in. 
After examining the elements in the browser's developer tools, in fact only the frame of a new slide is created with no content at all.

My expectation would have been that after correctly expanding _galleryLightbox.elements_ and then _reloading()_ the galleryLightbox object, this would then automatically update _galleryLightbox.slidesContainer_ as well (based on the newly inserted elements in _galleryLightbox.elements_) and thus show correct behavior when zooming.

What am I doing wrong?
Have I misunderstood the functionality of _glightbox.reload()_?
Or maybe the problem arises from the fact that I insert the new image _in galleryLightbox.elements[0]_ in the first step?
But actually it shouldn't matter, should it?
I am using _glightbox-3.2.0.min.js_

I kindly ask you for hints

**Comment by gingerchew** [comment]
Do you have a codepen or something similar we can look at? A reduced test case helps us out a lot
